common/src/main/scala/uk/ac/wellcome/platform/archive/common/operation/OutgoingPublisher.scala

```diff
@@ -1,42 +1,29 @@
 package uk.ac.wellcome.platform.archive.common.operation
 
-import java.util.UUID
-
 import io.circe.Encoder
-import uk.ac.wellcome.messaging.sns.{PublishAttempt, SNSWriter}
-import uk.ac.wellcome.platform.archive.common.ingests.operation.{
-  OperationCompleted,
-  OperationFailure,
-  OperationResult,
-  OperationSuccess
-}
+import uk.ac.wellcome.messaging.sns.SNSWriter
+import uk.ac.wellcome.platform.archive.common.ingests.operation.OperationResult
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class OutgoingPublisher(
   operationName: String,
   snsWriter: SNSWriter
 ) {
-  def send[R, O](
-    requestId: UUID,
-    result: OperationResult[R],
-  )(
-    transform: R => O
-  )(implicit
+  def sendIfSuccessful[R, O](result: OperationResult[R], outgoing: => O)(
+    implicit
     ec: ExecutionContext,
-    enc: Encoder[O]): Future[Unit] = result match {
-    case OperationSuccess(summary) =>
-      sendOutgoing(transform(summary)).map(_ => ())
-    case OperationFailure(_, _) =>
+    enc: Encoder[O]): Future[Unit] = {
+    if (result.isSuccessful) {
```

Comment:
You can replace this with a match and `OR` the two success type together - that way you can avoid the case to boolean mapping below.